### ISSUE TITLE
incorrect selection displayed when use as data range for one datepicker

### ISSUE THREAD

**Issue by Mattervispa** [Hacker0x01/react-datepicker#2549]
**--- Steps to Reproduce: ---**

use date picker as "Data range for one datepicker", see examples on reactdatepicker.com

select range from Nov. 1st to Nov. 10th

**--- Actual Result: ---**

date range from 1/11 to 10/11 is selected 
plus
**in any other month the last day of the data range (10th)**

**--- Expected Result: ---**

only range from 1/11 to 10/11 should be selected
 
**--- Workaround: ---**

n.a.

**--- Performance Impact: ---**

just a display issue, startDate and endDate are set correctly

**--- Environment Details: ---**

react-datepicker 3.30

**Comment by kenobivan** [comment]
Hi @Mattervispa,
I have the same issue. Did you find a solution?

Thanks!

**Comment by kenobivan** [comment]
This dirty CSS quick fix solved it for me:

.react-datepicker__month-container
  .react-datepicker__day--range-end.react-datepicker__day--keyboard-selected {
  background-color: #216ba5;
  color: #fff;
}

.react-datepicker__month-container .react-datepicker__day--keyboard-selected {
  background-color: white;
  color: #000;
}

.react-datepicker__month-container .react-datepicker__day--keyboard-selected:focus {
  background-color: #216ba5;
  color: #fff;
}

.react-datepicker__month-container .react-datepicker__day--keyboard-selected:hover {
  background-color: #f0f0f0;
}

**Comment by Mattervispa** [comment]
Hi @kenobivan 

Thank you for publishing the fix. The selectors are working fine for me. At the end I just use these ones:

```
.react-datepicker__month-container .react-datepicker__day--keyboard-selected {
  background-color: white;
  color: #000;
}
.react-datepicker__month-container .react-datepicker__day--keyboard-selected:hover {
  background-color: #f0f0f0;
  color: #000;
}
```

**Comment by Mattervispa** [comment]
still not fixed in 4.7.0, see [https://reactdatepicker.com/#example-date-range-for-one-datepicker](url)

**Comment by Nichon4** [comment]
You have to see highlight for .react-datepicker__day--keyboard-selected.
as workaround - you can disable keyboard navigation by [adding prop](https://reactdatepicker.com/#example-disable-keyboard-navigation)
